reactioncommerce/reaction
# Fix duped cart referenceId

Resolves #5073 
Impact: **critical**  
Type: **bugfix**

## Issue

`convertAnonymousCartToNewAccountCart` tries to create a new cart with referenceId that is the same as an existing cart but referenceId needs to be unique as defined in the Cart schema.

## Solution
Before creating a new cart, update the old cart (the anonymous cart) by removing the referenceId field so the new cart can be created with the same referenceId without having uniqueness issues.

## Breaking changes
None

## Testing
- Clear all documents in Cart collection.
- Make sure you are logged out in starterkit.
- Add an item to cart.
- Take note of the _id and referenceId of the cart in the database.
- Log in.
- See that a new account cart is created in the database with a different _id but same referenceId.
